testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/model/UserSessionProviderOfflineTest.java

```diff
@@ -112,10 +113,8 @@ public void testOfflineSessionsCrud(KeycloakSession session) {
             // Key is userSession ID, values are client UUIDS
             // Persist 3 created userSessions and clientSessions as offline
             ClientModel testApp = realm.getClientByClientId("test-app");
-            List<UserSessionModel> userSessions = currentSession.sessions().getUserSessions(realm, testApp);
-            for (UserSessionModel userSession : userSessions) {
-                offlineSessions.put(userSession.getId(), createOfflineSessionIncludeClientSessions(currentSession, userSession));
-            }
+            currentSession.sessions().getUserSessionsStream(realm, testApp).collect(Collectors.toList())
```

Comment:
Maybe we can replace collect + forEach with collectToMap + putAll
```suggestion
            offlineSessions.putAll(currentSession.sessions().getUserSessionsStream(realm, testApp)
                    .collect(Collectors.toMap(UserSessionModel::getId,
                            userSession -> createOfflineSessionIncludeClientSessions(currentSession, userSession))));
```

Reply:
It doesn't work in this case because createOfflineSessionIncludeClientSessions is creating new sessions while the stream is still being processed, which leads to concurrent modification exception. The prior approach collects the elements first and then attempts to add new offline sessions which now doesn't make changes to the stream source.

Reply:
I see. Thank you for the explanation.